### ISSUE TITLE
Room actor `manageConnections` permission not working

### ISSUE THREAD

**Issue by weeman1337** [matrix-org/matrix-hookshot#658]
Steps to reproduce:

- Create a room
- Set up `manageConnections` permissions with the room as actor
- With a user from that room
- Create another room
- Invite hookshot, make mod, create webhook
- Edit state and add transform function

Expected:

Allowed to do so

What happens:

Not allowed, hookshot resets the state event

```
ERROR 08:45:57:601 [ConnectionManager] User @test:example.com is disallowed to manage state for generic in !ASD:example.com

```

**Comment by Half-Shot** [comment]
Interesting, it's possible hookshot forgot to update the membership? Did you try restarting hookshot, did that make a difference?

**Comment by weeman1337** [comment]
No. Restarting doesn't solve it. Where int he code should it update the membership? I can try to debug this further.

**Comment by Half-Shot** [comment]
https://github.com/matrix-org/matrix-hookshot/blob/eee51d7c480e60766b40121f73b6573e517e109e/src/Config/Config.ts#L649-L665 is where we fill the membership cache

https://github.com/matrix-org/matrix-hookshot/blob/7e2ffdd2df5049ed2ee3a0245b179e44ff7a243e/src/Config/permissions.rs#L39-L40 is where we check the membership rules.
